### PR TITLE
[HEI-225] Dev/architecture v10 inner function

### PR DESCRIPTION
**Pull request by TonyKatkov89**

### CODE REVIEW

File: hypex/transformers/abstract.py [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
 
+    def execute(self, data: ExperimentData) -> ExperimentData:
+        data = data.copy(data=self.calc(data=data.ds))
+        return data
+
     @staticmethod
-    @abstractmethod
-    def _inner_function(data: Dataset, **kwargs) -> Any:
-        raise AbstractMethodError
+    def _list_unification(roles: Any):
```
Comment (VasilievaKA) [inline hypex/transformers/abstract.py:28]: leave this method into Adapter like: Adapter.to_list(value)

File: hypex/transformers/filters.py [hunk 2]
```diff
@@ -31,41 +31,44 @@ def __init__(
...
     ) -> Dataset:
-        target_roles = super()._list_unification(target_roles)
-        if type_filter:
-            addressable_columns = data.search_columns(
-                roles=target_roles, search_types=[float, int, bool]
-            )
-        else:
-            addressable_columns = data.search_columns(target_roles)
-        for column in addressable_columns:
+        target_cols = super(CVFilter, CVFilter)._list_unification(target_cols)
```
Comment (VasilievaKA) [inline hypex/transformers/filters.py:43]: you can write just super()._list_unification

File: hypex/transformers/filters.py [hunk 3]
```diff
@@ -31,41 +31,44 @@ def __init__(
...
+        for column in target_cols:
             cv = data[column].coefficient_of_variation()
             drop = False
-            if self.upper_bound and cv > self.upper_bound:
+            if upper_bound and cv > upper_bound:
                 drop = True
-            if self.lower_bound and cv < self.lower_bound:
+            if lower_bound and cv < lower_bound:
```
Comment (VasilievaKA) [inline hypex/transformers/filters.py:49]: merge this two checks into one throw `or`

File: hypex/transformers/shuffle.py [hunk 4]
```diff
@@ -10,20 +10,17 @@ def __init__(
         random_state: Optional[int] = None,
         key: Any = "",
     ):
-        super().__init__( key)
+        super().__init__(key)
         self.random_state = random_state
 
+    @staticmethod
+    def _inner_function(data: Dataset, random_state: Optional[int] = None) -> Any:
```
Comment (VasilievaKA) [inline hypex/transformers/shuffle.py:17]: return Dataset but not Any

File: hypex/transformers/shuffle.py [hunk 5]
```diff
@@ -10,20 +10,17 @@ def __init__(
...
+        result = data.shuffle(random_state=random_state)
+        return result
```
Comment (VasilievaKA) [inline hypex/transformers/shuffle.py:19]:
```python
return data.shuffle(random_state=random_state)
```

**Review by VasilievaKA — CHANGES_REQUESTED** [review]
1️⃣0️⃣❗

File: hypex/transformers/filters.py (hunk 2 above)
Comment (TonyKatkov89) [reply hypex/transformers/filters.py:43]: By some reason it does not work, but anyways it is replaced with adapter method.

File: hypex/dataset/abstract.py [hunk 6]
```diff
@@ -161,3 +161,6 @@ def to_json(self, filename: Optional[str] = None):
     @property
     def backend(self):
         return self._backend
+
+    def get_values(self, row: FieldKeyTypes = None, column: FieldKeyTypes = None) -> Any:
```
[the remark this answers is not in the document]
Comment (TonyKatkov89) [reply hypex/dataset/abstract.py]: fixed

File: hypex/dataset/backends/abstract.py [hunk 7]
```diff
@@ -245,6 +245,9 @@ def agg(self, func) -> Any:
         raise AbstractMethodError
 
     @abstractmethod
+    def get_values(self, row: FieldKeyTypes = None, column: FieldKeyTypes = None) -> Any:
```
[the remark this answers is not in the document]
Comment (TonyKatkov89) [reply hypex/dataset/backends/abstract.py]: fixed

File: hypex/dataset/dataset.py [hunk 8]
```diff
@@ -304,12 +304,11 @@ def apply(
         axis: int = 0,
         **kwargs,
     ) -> "Dataset":
-        return Dataset(
-            data=self._backend.apply(func=func, axis=axis, **kwargs).rename(
-                list(role.keys())[0]
-            ),
-            roles=role,
-        )
+            data = self._backend.apply(func=func, axis=axis, column_name=list(role.keys())[0], **kwargs)
```
[the remark this answers is not in the document]
Comment (TonyKatkov89) [reply hypex/dataset/dataset.py]: ****

[the remark this answers is not in the document]
Comment (TonyKatkov89) [reply hypex/dataset/dataset.py]: fixed

File: hypex/dataset/dataset.py [hunk 9]
```diff
@@ -436,7 +435,10 @@ def value_counts(
             normalize=normalize, sort=sort, ascending=ascending, dropna=dropna
         )
         t_roles = self.roles
-        t_roles["count"] = StatisticRole()
+        if normalize:
```
[the remark this answers is not in the document]
Comment (TonyKatkov89) [reply hypex/dataset/dataset.py]: fixed

File: hypex/transformers/abstract.py (hunk 1 above)
Comment (TonyKatkov89) [reply hypex/transformers/abstract.py]: fixed

File: hypex/transformers/shuffle.py (hunk 4 above)
Comment (TonyKatkov89) [reply hypex/transformers/shuffle.py]: fixed

File: hypex/transformers/shuffle.py (hunk 5 above)
Comment (TonyKatkov89) [reply hypex/transformers/shuffle.py]: fixed

**Review by VasilievaKA — APPROVED** [review]

**Pull request merged**